app/sockets/service.py

```diff
@@ -101,7 +101,44 @@ def create_conversation(conversation_id, creator, participants):
                 result.append(participant)
                 
         return result
-    
+
+    @staticmethod
+    def remove_identifier(email):
+        """
+        Removes the provided email from the internal mapping
+        of users to socket IDs
+        
+        :email        the email address of a user
+        """
+        
+        if email in SocketService._sockets:
+            del SocketService._sockets[email]
+
+    @staticmethod
+    def leave_conversation(socket_id, conversation_id, participant):
+        """
+        Removes a participant from a conversation
+        
+        :socket_id        the 'conversation' socket ID for the participant
+        :conversation_id  a unique ID for a specific conversation
+        :participant      the email address of the participant
+        """
+        if socket_id in SocketService._sockets.values():
+            socketio = current_app.extensions['socketio']
+            socketio.server.leave_room(socket_id, conversation_id, namespace='/conversation')
```

Comment:
You should retrieve `socket_id` from the `_sockets` dictionary directly (the current if statement will always fail). 

You'll probably want something like "if participant is in the sockets dictionary, get the socket ID of the conversation namespace and store it in some variable. Then, call leave_room."